distributed/scheduler.py

```diff
@@ -5562,7 +5642,8 @@ def send_task_to_worker(self, worker, ts: TaskState, duration: double = -1):
     def handle_uncaught_error(self, **msg):
         logger.exception(clean_exception(**msg)[1])
 
-    def handle_task_finished(self, key=None, worker=None, **msg):
+    @stimulus_handler(sync=True)
+    def handle_task_finished(self, key=None, worker=None, stimulus_id=None, **msg):
```

Comment:
I tried typing some of these functions, but one of the reasons they are untyped is to avoid conflicts between cython and mypy typing (cython has precedence AFAIK).